lib/segment/src/common/rocksdb_buffered_delete_wrapper.rs

```diff
@@ -19,6 +21,16 @@ pub struct DatabaseColumnScheduledDeleteWrapper {
     deleted_pending_persistence: Mutex<HashSet<Vec<u8>>>,
 }
 
+impl Clone for DatabaseColumnScheduledDeleteWrapper {
+    fn clone(&self) -> Self {
+        let deleted_pending_persistence = self.deleted_pending_persistence.lock().clone();
+        Self {
+            db: self.db.clone(),
+            deleted_pending_persistence: Mutex::new(deleted_pending_persistence),
+        }
+    }
+}
```

Comment:
The current usages of `.clone()` are fine (`immutable_map_index.rs` and `immutable_numeric_index.rs`), but this `Clone` implementation looks like a potential footgun to me. (think of two clones with diverging hashsets)

I'd prefer to do the ether:
1. Put it under `Arc` (e.g. `Arc<Mutex<HashSet<Vec<u8>>>>`).
2. Or rewrite the code to not to use `.clone()`. Not in this PR, a `// TODO` or `#[deprecated]` would suffice there.

Reply:
It's possible to avoid using this `Clone`. Clone is called in mutable-into-immutable conversion. We can avoid cloning of `DatabaseColumnScheduledDeleteWrapper` and create a new `DatabaseColumnScheduledDeleteWrapper` instead